### PR TITLE
Revise README.md for clarity

### PR DESCRIPTION
**Pull request by Lupie**
- Fix typos
- Clarify what Provide does
- Reorganize content

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -109,6 +83,30 @@ c.Invoke(func(map map[string]int, slice []int, array [2]string) {
 })
 ```
 
+### Provide a Constructor
```
Comment (glibsm) [inline README.md]: constructor is the main use case and I'd love to keep it as one of the first sections

File: README.md [hunk 2]
```diff
@@ -32,48 +32,23 @@ Objects in the container are identified by their `reflect.Type` and **everything
...
-// have not been provided. Constructors (or instances) first
-// have to be provided to the dig container before it is able
-// to create a shared singleton instance of *Type3
-```
-
-### Provide an Object
-
-Registering an object directly is a shortcut to register something that
-has no dependencies.
+As a shortcut for objects without dependencies, register it directly.
```
Comment (glibsm) [inline README.md]: I think it's weird to talk about "a shortcut" before explaining what the full version is

File: README.md (hunk 1 above)
Comment (Lupie) [inline README.md]: No prob. I just want the short list to match the order of the longer explanations. I'll adjust the short list instead.

File: README.md (hunk 2 above)
Comment (Lupie) [inline README.md]: Is this problem solved if constructor comes first?

**Comment by glibsm** [comment]
thank you!

**Pull request merged**